datasets/fr/tresor/crawler.py

```diff
@@ -101,6 +126,70 @@ def parse_identification(
                 wallet.add("holder", entity.id)
                 context.emit(wallet)
 
+    return True
+
+
+def parse_identification(
+    context: Context,
+    entity: Entity,
+    value: Dict[str, str],
+):
+    comment = value.pop("Commentaire")
+    content = value.pop("Identification")
+    full = f"{comment}: {content}".strip(SEPARATORS)
+
+    # First, check if we have a lookup for the whole identification field, overriding the segment splitting logic
+    if apply_identification_lookup(
+        context, context.get_lookup("identification_full"), entity, full
+    ):
+        return
+
+    segments = parse_split(context, entity, full)
+    for segment in segments:
+        if apply_identification_lookup(
+            context, context.get_lookup("identification_segment"), entity, segment
+        ):
+            continue
+
+        for key, propname in TEXT_KEYS.items():
+            if segment.lower().startswith(key.lower()) and propname is not None:
+                value = segment[len(key) :].strip(SEPARATORS)
+
+                if propname == "kppCode":
+                    entity.add_cast(
+                        "Company", propname, value, lang="fra", original_value=segment
+                    )
+                elif propname == "incorporationDate":
+                    h.apply_date(entity, propname, value)
+                else:
+                    entity.add(propname, value, lang="fra", original_value=segment)
+
+                # We found what this key means, no need to try other keys.
+                break
+        else:
+            # When processing this as part of daily issues, here are some options:
+            # a) add a new key to TEXT_KEYS if it seems like it might be/become common
+            # b) add a new lookup to identification_segment to map the segment
+            # c) add a new lookup to identification_full if the splitting failed or you want to match the full segment
+            #      for some other reason
+            context.log.warning(
+                f'Failed to parse identification segment: "{segment}"',
+                segment=segment,
+                full=full,
+            )
+
+    # Special handling for Russian tax/registration numbers, which is most companies on the list
+    if followthemoney.model.get("LegalEntity") in entity.schema.schemata:
+        registration_number = entity.first("registrationNumber")
+        if registration_number and rigour.ids.OGRN.is_valid(registration_number):
+            entity.add("ogrnCode", registration_number)
+            entity.remove("registrationNumber", registration_number)
+
+        tax_number = entity.first("taxNumber")
+        if tax_number and rigour.ids.INN.is_valid(tax_number):
+            entity.add("innCode", tax_number)
+            entity.remove("taxNumber", tax_number)
```

Comment:
Does this maybe fit better with the `if propname == "kppCode":` sort of stuff or does that make the code more awkward in a way I don't quickly see? So that the special cases are together and perhaps more importantly, we don't do the awkward add + remove + add

Reply:
The advantage here is that we also catch those registrationNumbers that we mapped manually --- do we want that even?

Reply:
I went with what you suggested.